tests/storage/blockvolume_test.py

```diff
@@ -196,3 +198,15 @@ def test_repair_capacity(self, capacity, virtual_size, expected_capacity):
 
             vol.updateInvalidatedSize()
             assert vol.getMetadata().capacity == expected_capacity
+
+    @permutations([
+        # format, prealloc, can_reduce
+        (sc.RAW_FORMAT, sc.PREALLOCATED_VOL, False),  # invalid format and type
+        (sc.COW_FORMAT, sc.PREALLOCATED_VOL, False),  # invalid type
+        (sc.RAW_FORMAT, sc.SPARSE_VOL, False),  # invalid format
+        (sc.COW_FORMAT, sc.SPARSE_VOL, True),  # all OK
```

Comment:
I forgot to submit this comment last time - we don't test anything invalid here, all
arguments are valid, so the comments are misleading.

We can describe the arguments better like this:

        # Raw or preallocated disks cannot be reduced.
        (sc.RAW_FORMAT, sc.PREALLOCATED_VOL, False),
        (sc.COW_FORMAT, sc.PREALLOCATED_VOL, False),
        # Cow sparse can be reduced.
        (sc.COW_FORMAT, sc.SPARSE_VOL, True),

We don't need to test:

        (sc.RAW_FORMAT, sc.SPARSE_VOL, False),

Because this is an invalid combination, we cannot have such disks. You can add a comment about this if you think this is not clear enough. It may help new contributors.